optuna/samplers/nsgaii/_sampler.py

```diff
@@ -410,29 +410,46 @@ def after_trial(
 
 
 def _calc_crowding_distance(population: List[FrozenTrial]) -> DefaultDict[int, float]:
+
     manhattan_distances: DefaultDict[int, float] = defaultdict(float)
+    if len(population) == 0:
+        return manhattan_distances
+
     for i in range(len(population[0].values)):
         population.sort(key=lambda x: cast(float, x.values[i]))
 
-        v_min = population[0].values[i]
-        v_max = population[-1].values[i]
-        assert v_min is not None
-        assert v_max is not None
-
-        width = v_max - v_min
-        if width == 0:
+        # If population have the same values[i], ignore that value.
+        if population[0].values[i] == population[-1].values[i]:
             continue
 
-        manhattan_distances[population[0].number] = float("inf")
-        manhattan_distances[population[-1].number] = float("inf")
+        vs = (
+            [-float("inf")]
+            + [cast(List[float], population[j].values)[i] for j in range(len(population))]
+            + [float("inf")]
+        )
+
+        # Smallest finite value.
+        v_min = next(x for x in vs if x != -float("inf"))
 
-        for j in range(1, len(population) - 1):
-            v_high = population[j + 1].values[i]
-            v_low = population[j - 1].values[i]
-            assert v_high is not None
-            assert v_low is not None
+        # Largest finite value.
+        v_max = next(x for x in reversed(vs) if x != float("inf"))
 
-            manhattan_distances[population[j].number] += (v_high - v_low) / width
+        width = v_max - v_min
+        if width <= 0:
+            # width == 0 or width == -inf
+            width = 1.0
+
+        # subtract_nonfinite(inf, inf) == 0
+        def subtract_nonfinite(x: float, y: float) -> float:
+            if x == y:
+                return 0
+            else:
+                return x - y
+
+        for j in range(len(population)):
+            manhattan_distances[population[j].number] += (
+                subtract_nonfinite(vs[j + 2], vs[j]) / width
+            )
```

Comment:
A nit, but I think there is no need to define this as a function since used only once.
```suggestion
        for j in range(len(population)):
            # inf - inf and (-inf) - (-inf) is considered to be zero.
            gap = 0.0 if vs[j] == vs[j + 2] else vs[j + 2] - vs[j]
            manhattan_distances[population[j].number] += gap / width
```